Index: lib/quota/quotaio.c
===================================================================
--- lib/quota/quotaio.c.orig	2012-03-18 19:18:33.000000000 +0100
+++ lib/quota/quotaio.c	2012-04-13 12:57:21.261673322 +0200
@@ -142,8 +142,8 @@ errcode_t quota_inode_truncate(ext2_fils
 		return err;
 
 	inode.i_dtime = fs->now ? fs->now : time(0);
-	if (!ext2fs_inode_has_valid_blocks2(fs, &inode))
-		return 0;
+	if (!(err = ext2fs_inode_has_valid_blocks2(fs, &inode)) )
+		return err;
 
 	ext2fs_block_iterate3(fs, ino, BLOCK_FLAG_READ_ONLY, NULL,
 			      release_blocks_proc, NULL);
